src/process/process_set.cu

```diff
@@ -0,0 +1,135 @@
+#include <iostream>
+
+namespace micm {
+    namespace cuda {
+      // flipped memory layout
+    __global__ void AddForcingTerms_kernel(
+        double* rate_constants, 
+        double* state_variables, 
+        double* forcing, 
+        int ngrids,
+        int nrxns, 
+        int nspecs, 
+        size_t* number_of_reactants_, 
+        size_t* reactant_ids_, 
+        size_t* number_of_products_, 
+        size_t* product_ids_, 
+        double* yields_)
+    {
+      //define thread index 
+      size_t tid = blockIdx.x * blockDim.x + threadIdx.x; 
+      size_t react_id_offset, prod_id_offset, yield_offset;
+
+      if (tid < ngrids){
+         react_id_offset = 0;
+	 prod_id_offset = 0;
+	 yield_offset = 0;
+         for (std::size_t i_rxn = 0; i_rxn < nrxns; ++i_rxn)
+         {
+           double rate = rate_constants[i_rxn*ngrids+tid];
+           size_t index;
+
+           for (std::size_t i_react = 0; i_react < number_of_reactants_[i_rxn]; ++i_react)
+             rate *= state_variables[reactant_ids_[react_id_offset+i_react]*ngrids+tid];
+           for (std::size_t i_react = 0; i_react < number_of_reactants_[i_rxn]; ++i_react){
+             forcing[reactant_ids_[react_id_offset+i_react]*ngrids+tid] -= rate;
+	   }
+           for (std::size_t i_prod = 0; i_prod < number_of_products_[i_rxn]; ++i_prod){
+             index = product_ids_[prod_id_offset+i_prod]*ngrids+tid;
+             forcing[index] += yields_[yield_offset+i_prod] * rate;
```

Comment:
```suggestion

           for (std::size_t i_react = 0; i_react < number_of_reactants_[i_rxn]; ++i_react)
             rate *= state_variables[reactant_ids_[react_id_offset+i_react]*ngrids+tid];
           for (std::size_t i_react = 0; i_react < number_of_reactants_[i_rxn]; ++i_react){
             forcing[reactant_ids_[react_id_offset+i_react]*ngrids+tid] -= rate;
	   }
           for (std::size_t i_prod = 0; i_prod < number_of_products_[i_rxn]; ++i_prod){
             size_t index = product_ids_[prod_id_offset+i_prod]*ngrids+tid;
             forcing[index] += yields_[yield_offset+i_prod] * rate;
```

If this is only used here, which it seems to be, I think it's better to move the declaration next to the initialization so that no bugs creep in in the future from uninitialized variables

Reply:
Thanks. That sounds good to me. @qinatan can you make the code change here?

Reply:
done.